SofaKernel/modules/Sofa.Geometry/src/sofa/geometry/Triangle.h

```diff
@@ -31,6 +33,19 @@ struct Triangle
     static const sofa::Size NumberOfNodes = 3;
 
     Triangle() = default;
+
+    template<typename Node,
+        typename T = std::decay_t<decltype(*std::begin(std::declval<Node>()))>,
+        typename = std::enable_if_t<std::is_scalar_v<T>>>
+        static constexpr auto area(const Node& n0, const Node& n1, const Node& n2)
+    {
+        const auto a = sofa::geometry::Edge::length(n0, n1);
+        const auto b = sofa::geometry::Edge::length(n0, n2);
+        const auto c = sofa::geometry::Edge::length(n1, n2);
+
+        const auto area = static_cast<T>(0.25) * std::sqrt((a + b + c) * (-a + b + c) * (a - b + c) * (a + b - c));
+        return area;
```

Comment:
```suggestion
        const auto a = n1 - n0;
        const auto b = n2 - n0;

        const auto area = static_cast<T>(0.5) * std::abs(sofa::type::cross(a, b));
        return area;
```
Your initial implementation uses Heron's formula and involved 4 square roots. If you work with squared length, it leads to only one square root. 
But for the 2d and 3d case, I think computing the half of the parallelogram is faster as it does not involve any square root. My suggestion is for the 3d case. The code may require some changes for the 2d case, but the principle is the same.